src/particles.cpp

```diff
@@ -50,4 +53,11 @@ void ParticleParameters::deSerialize(std::istream &is, u16 protocol_ver)
 	animation.deSerialize(is, 6); /* NOT the protocol ver */
 	glow               = readU8(is);
 	object_collision   = readU8(is);
+	// This is kinda awful
```

Comment:
How about adding a version number on top that can be used reliably beginning from protocol version 40?

Reply:
The awful thing here is explicitly needing to check for EOF, other than that it's not different from other `try { ... } catch(PacketError) {}` code.
So I don't think it's good to bump the protocol version when it isn't strictly necessary.

Reply:
The difference to `PacketError` is that the variables aren't overwritten when EOF is reached - here it requires this workaround.
I suggested particle-versioning to get around protocol bumps in the future and to solve these checks without the "awful" part. The protocol version can be bumped a lot; it's a `u16`.

Reply:
Right. When `readU8`/`readU16` are replaced with an API that properly throws exceptions in the future then the awful code vanishes and the extra version number would be wasted work.
So IMO this shouldn't be done.